src/cryptocom/exchange/api.py

```diff
@@ -76,15 +95,38 @@ def _sign(self, path, data):
         ).hexdigest()
         return data
 
+    def set_limit(self, url):
+
+        if not(url in self.limits.keys()):
+            if url.startswith('private'):
+                rate_limit, period = 3, 0.1
+            
+            elif url.startswith('public'):
+                rate_limit, period = 100, 1
+
+            else:
+                raise ApiError(f'Wrong path: {url}')
+
+        else:
+            rate_limit, period = self.limits[url]
+        
+        self.limiter.max_rate = rate_limit
+        self.limiter.time_period = period
+
     async def request(self, method, path, params=None, data=None, sign=False):
         original_data = data
         timeout = aiohttp.ClientTimeout(total=self.timeout)
+
+        if not (path == self.last_request):
+            self.set_limit(path)
```

Comment:
if you done for example 2-3 parallel requests, you will trigger set_limit, set_limit, set_limit and then your limit will be cleared. You need to create a pool of aioratelimiters and grab from pool each one per matched request

Reply:
limiter = self.limits[request.path] or def get_limiter(self, path): if we matched path return limiter